uniffi_bindgen/src/interface/types/mod.rs

```diff
@@ -242,73 +258,17 @@ impl TypeUniverse {
     }
 
     /// Iterator over all the known types in this universe.
-    pub fn iter_known_types(&self) -> impl Iterator<Item = Type> + '_ {
-        self.all_known_types.iter().cloned()
+    pub fn iter_known_types(&self) -> impl Iterator<Item = &Type> {
+        self.all_known_types.iter()
     }
 }
 
 /// An abstract type for an iterator over &Type references.
 ///
 /// Ideally we would not need to name this type explicitly, and could just
-/// use an `impl Iterator<Item=&Type>` on any method that yields types.
-/// Unfortunately existential types are not currently supported in trait method
-/// signatures, so for now we hide the concrete type behind a box.
+/// use an `impl Iterator<Item = &Type>` on any method that yields types.
 pub type TypeIterator<'a> = Box<dyn Iterator<Item = &'a Type> + 'a>;
```

Comment:
I kept this type for most places it was previously used because there are a few places where different functions returning it are returned from `match` arms and since the underlying iterators aren't the same, `impl Iterator` wouldn't actually work.

I kept most of the comment on the type though because ideally with something like `enum impl Trait` (or just further refactoring of `uniffi_bindgen`), this type could still be removed eventually.